eppoclient/configresponse.go

```diff
@@ -20,17 +26,39 @@ type flagConfiguration struct {
 	Variations    map[string]variation `json:"variations"`
 	Allocations   []allocation         `json:"allocations"`
 	TotalShards   int64                `json:"totalShards"`
-}
-
-func (flag *flagConfiguration) Precompute() {
+	// Cached Variations parsed according to `VariationType`.
+	//
+	// Types are as follows:
+	// - STRING -> string
+	// - NUMERIC -> float64
+	// - INTEGER -> int64
+	// - BOOLEAN -> bool
+	// - JSON -> jsonVariationValue
+	ParsedVariations map[string]interface{} `json:"-"`
+}
+
+func (flag *flagConfiguration) precompute() {
 	for i := range flag.Allocations {
-		flag.Allocations[i].Precompute()
+		flag.Allocations[i].precompute()
+	}
+
+	flag.ParsedVariations = make(map[string]interface{}, len(flag.Variations))
+	for i := range flag.Variations {
+		value, err := flag.VariationType.parseVariationValue(flag.Variations[i].Value)
+		if err == nil {
+			flag.ParsedVariations[i] = value
+		}
```

Comment:
are we ignoring the `else` case here? presumably we want to at the very least log the error I assume

Reply:
Yes.

The application logger is not available in this function and I want to avoid piping it through client -> configuration store -> multiple precomputes for something that should never happen (bad server configuration).

We log an error in evaluation function though, so it doesn't go completely unnoticed.